docs/user/dk_bench.md

```diff
@@ -0,0 +1,183 @@
+
+Additionally, there was no way to bring a custom evaluation data set to run on InstructLab trained models.
+
+DK-Bench (Domain-Knowledge Bench) provides the ability to bring custom evaluation questions and score the models answers on a sliding scale.
+
+## Description
+
+DK-Bench provides a flexible way for users to bring custom evaluation questions, reference answers, and collect a models answer to the questions.
+The responses to each of questions is given a score from 1-5 by a judge model.
+The current default judge model is GPT-4o by OpenAI.
+
+### User Input
+
+To run DK-Bench, users must provide a `.jsonl` file with every question they wish to ask a model to answer and evaluate.
+
+Each line must be valid JSON, and contain the entries `user_input` for the question, and `reference` for the ground truth answer to the question.
+
+Here is an example of a `.jsonl` file that can be used in DK-Bench with 2 questions:
+
+```json
+{"user_input":"What is the capital of Canada?","reference":"The capital of Canada is Ottawa."}
+{"user_input":"What is the capital of Mexico?","reference":"The capital of Mexico is Mexico City."}
+```
+
+### Rubric
+
+Each response given by a model is compared to the reference answer and graded on the following 1 - 5 scale by the judge model:
+
+| Score | Criteria |
+| ---------------- | -------- |
+| 1 | The response is entirely incorrect, irrelevant, or does not align with the reference in any meaningful way |
+| 2 | The response partially matches the reference but contains major errors, significant omissions, or irrelevant information |
+| 3 | The response aligns with the reference overall but lacks sufficient detail, clarity, or contains minor inaccuracies |
+| 4 | The response is mostly accurate, aligns closely with the reference, and contains only minor issues or omissions |
+| 5 | The response is fully accurate, completely aligns with the reference, and is clear, thorough, and detailed |
+
+### Output
+
+After running DK-Bench, users will see a report with the question by question score, the average, and the total for their run.
+
+```shell
+# DK-BENCH REPORT
+
+## MODEL: granite-7b-lab
+
+Question #1:     5/5
+Question #2:     5/5
+Question #3:     5/5
+Question #4:     5/5
+Question #5:     2/5
+Question #6:     3/5
+Question #7:     2/5
+Question #8:     3/5
+Question #9:     5/5
+Question #10:     5/5
+----------------------------
+Average Score:   4.00/5
+Total Score:     40/50
+
+Responses and scores written to:
+~/.local/share/instructlab/internal/eval_data/dk_bench/granite-7b-lab/results_2025-02-04T16:18:02.337010.jsonl
+```
+
+The original question, the responses a model gave for that specific question, the reference answer, the score of the judge model are all output to an output file.
+
+The output file could be `.jsonl` (the default and seen below), `.csv`, and `.xlsx`. This can be set using the `--output-file-formats` flag or in the `dk_bench` section of the configuration file.
```

Comment:
```suggestion
The output file can be `.jsonl` (the default and seen below), `.csv`, or `.xlsx`. This can be set using the `--output-file-formats` flag or in the `dk_bench` section of the configuration file.
```